Swiftfin tvOS/Views/SelectUserView/Components/ServerSelectionMenu.swift

```diff
@@ -43,33 +40,97 @@ extension SelectUserView {
+                        Button {
+                            router.route(to: \.editServer, selectedServer)
+                        } label: {
+                            HStack {
+                                L10n.editServer.text
+                                    .font(.headline)
+
+                                Spacer()
+
+                                Image(systemName: "server.rack")
                             }
                         }
                     }
-                    .font(.body.weight(.semibold))
-                    .foregroundStyle(Color.primary)
                 }
+                Section(L10n.selectedServer) {
+                    if viewModel.servers.keys.count > 1 {
+                        Button {
+                            serverSelection = .all
+                            router.popLast()
+                        } label: {
+                            HStack {
+                                L10n.allServers.text
+                                    .font(.headline)
+
+                                Spacer()
+
+                                if serverSelection == .all {
+                                    Image(systemName: "checkmark.circle.fill")
+                                }
+                            }
+                        }
+                    }
+                    ForEach(viewModel.servers.keys.reversed()) { server in
+                        Button {
+                            serverSelection = .server(id: server.id)
+                            router.popLast()
+                        } label: {
+                            HStack {
+                                VStack(alignment: .leading) {
+                                    Text(server.name)
+                                        .lineLimit(1)
+                                        .font(.headline)
+
+                                    Text(server.currentURL.absoluteString)
+                                        .lineLimit(1)
+                                        .font(.subheadline)
+                                        .foregroundColor(.secondary)
+                                }
+
+                                Spacer()
+
+                                if selectedServer == server {
+                                    Image(systemName: "checkmark.circle.fill")
+                                }
+                            }
+                        }
+                    }
+                }
+            } label: {
+                Group {
+                    switch serverSelection {
+                    case .all:
+                        Label(L10n.allServers, systemImage: "person.2.fill")
+                    case let .server(id):
+                        if let server = viewModel.servers.keys.first(where: { $0.id == id }) {
+                            Label(server.name, systemImage: "server.rack")
+                        }
+                    }
+                }
```

Comment:
This should also have the menu indicator like on iOS:

https://github.com/jellyfin/Swiftfin/blob/b0583125f78f791e3a053619abae370d7c989089/Swiftfin/Views/SelectUserView/Components/ServerSelectionMenu.swift#L93

Reply:
Mirroring iOS is a a little off on the centering:

<img width="478" alt="Screenshot 2024-12-17 at 14 32 21" src="https://github.com/user-attachments/assets/9adbb021-0869-48c8-8457-a2e83e3d0429" />

```
                HStack {
                    switch serverSelection {
                    case .all:
                        // Image(systemName: "person.2.fill")
                        Label(L10n.allServers, systemImage: "person.2.fill")
                    case let .server(id):
                        if let server = viewModel.servers.keys.first(where: { $0.id == id }) {
                            // Image(systemName: "server.rack")
                            Label(server.name, systemImage: "server.rack")
                        }
                    }

                    Image(systemName: "chevron.up.chevron.down")
                        .foregroundStyle(.secondary)
                        .font(.subheadline.weight(.semibold))
                }
```

If I change it to this, it centers a bit better:

```
                HStack {
                    switch serverSelection {
                    case .all:
                        Image(systemName: "person.2.fill")
                        Text(L10n.allServers)//, systemImage: "person.2.fill")
                    case let .server(id):
                        if let server = viewModel.servers.keys.first(where: { $0.id == id }) {
                            Image(systemName: "server.rack")
                            Text(server.name)//, systemImage: "server.rack")
                        }
                    }

                    Image(systemName: "chevron.up.chevron.down")
                        .foregroundStyle(.secondary)
                        .font(.subheadline.weight(.semibold))
                }
```

<img width="481" alt="Screenshot 2024-12-17 at 14 31 38" src="https://github.com/user-attachments/assets/ceb92486-7996-4ab6-8679-24f57274a200" />

Reply:
Ah, the issue is due to the spacing differences between the `Label` contents and between `HStack` contents. However the server and disclosure icons should more closely hug the text, so maybe setting the `HStack(spacing:)` to something sensible would be better.